modules/scope3/rtd/README.md

```diff
@@ -0,0 +1,140 @@
+          "auth_key": "your-scope3-auth-key",
+          "timeout_ms": 1000
+        }
+      }
+    }
+  }
+}
+```
+
+## Environment Variables
+- `SCOPE3_API_KEY`: Your Scope3 API key for authentication
+
+## Features
+- Fetches real-time audience segments from Scope3
+- Adds segments to bid request targeting data
+- Thread-safe segment storage during auction
+- Configurable timeout and endpoint
+- Graceful error handling (doesn't fail auctions on API errors)
+- Integration with LiveRamp ATS for enhanced targeting
+
+## Dependencies
+This module can work with LiveRamp ATS to enhance targeting with RampID data. If you're using LiveRamp ATS, ensure it runs before the Scope3 module in your execution plan to populate user identifiers.
+
+### Integration with LiveRamp ATS
+When using both LiveRamp ATS and Scope3 RTD modules, configure execution order so LiveRamp runs first:
+
+```yaml
+host_execution_plan:
+  endpoints:
+    /openrtb2/auction:
+      stages:
+        raw_auction_request:
+          groups:
+            # Group 1: LiveRamp ATS runs first to populate RampID
+            - timeout: 1000
+              hook_sequence:
+                - module_code: "liveramp.ats"
+                  hook_impl_code: "liveramp-fetch"
+            # Group 2: Scope3 runs after LiveRamp, can use RampID
+            - timeout: 2000
+              hook_sequence:
+                - module_code: "scope3.rtd"
+                  hook_impl_code: "scope3-fetch"
+        processed_auction_request:
+          groups:
+            - timeout: 5
+              hook_sequence:
+                - module_code: "scope3.rtd"
+                  hook_impl_code: "scope3-targeting"
+```
+
+Alternative configuration (same group, sequential execution):
+```yaml
+raw_auction_request:
+  groups:
+    - timeout: 3000
+      hook_sequence:
+        # LiveRamp runs first
+        - module_code: "liveramp.ats"
+          hook_impl_code: "liveramp-fetch"
+        # Scope3 runs second, can access RampID
+        - module_code: "scope3.rtd"
+          hook_impl_code: "scope3-fetch"
+```
+
+## User Identifier Integration
+The module automatically detects and includes user identifiers when available in the bid request. This includes support for:
+
+### LiveRamp Integration Options
+1. **ATS Sidecar** (when LiveRamp module runs first):
+   - `user.ext.eids[]` array with `source: "liveramp.com"`
+   - `user.ext.rampid` field
+
+2. **ATS Envelope** (when no sidecar is available):
+   - `user.ext.liveramp_idl` - Primary ATS envelope location
+   - `user.ext.ats_envelope` - Alternative envelope location
+   - `ext.liveramp_idl` - Request-level envelope
+
+### How It Works
+- **With Sidecar**: LiveRamp decrypts the envelope and populates RampID, which Scope3 receives as a resolved identifier
+- **Without Sidecar**: The encrypted ATS envelope is forwarded directly to Scope3 API, allowing Scope3 to decrypt it (if they're an authorized LiveRamp partner)
```

Comment:
i think scope3 should know "if they're an authorized LiveRamp partner" or not, this shouldnt be mysterious to you